cibuildwheel/action.yml

```diff
@@ -0,0 +1,169 @@
+      shell: bash
+
+    - name: Install pipx
+      run: |
+        python3 -m pip install pipx
+        python3 -m pipx ensurepath --force
+      shell: bash
+
+    - name: Create conditional env vars with bash
+      run: |
+        arch=$(uname -m)
+
+        # set arch according to runner arch
+        echo "CIBW_ARCHS_LINUX=${arch}" >> "${GITHUB_ENV}"
+
+        cibw_before_all_ssh=""
+        # add some commands to set up an ssh agent for cloning private repos like cumlprims_mg and cugraph-ops
+        if [ "${{ inputs.cumlprims_mg-private-deploy-key }}" != '' ] || [ "${{ inputs.cugraph-ops-private-deploy-key }}" != '' ]; then
+          private_key="${{ inputs.cumlprims_mg-private-deploy-key }}${{ inputs.cugraph-ops-private-deploy-key }}"
+          # multiline env var: https://github.com/orgs/community/discussions/24952
+          echo 'PRIVATE_SSH_KEY<<EOF' >> "${GITHUB_ENV}"
+          echo "$private_key" >> "${GITHUB_ENV}"
+          # add a trailing newline to the ssh key: https://stackoverflow.com/questions/52109775/gitlab-ci-ssh-key-invalid-format/59595773#59595773
+          echo 'EOF' >> "${GITHUB_ENV}"
+          # forward multline env var to cibuildwheel container
+          echo "CIBW_ENVIRONMENT_PASS_LINUX=PRIVATE_SSH_KEY" >> "${GITHUB_ENV}"
+          # add forwarded multline env var later on, by escaping immediate expansion
+          cibw_before_all_ssh="echo \"\$PRIVATE_SSH_KEY\" > /private-deploy-key && chmod 0400 /private-deploy-key &&"
+        fi
+
+        cibw_before_all=""
+        auditwheel_repair_command=""
+        if [ "${arch}" == "x86_64" ]; then
+          cibw_before_all="${{ inputs.cibw-before-all-amd64 }}"
+          auditwheel_repair_command="${{ inputs.auditwheel-repair-command-amd64 }}"
+        elif [ "${arch}" == "aarch64" ]; then
+          cibw_before_all="${{ inputs.cibw-before-all-arm64 }}"
+          auditwheel_repair_command="${{ inputs.auditwheel-repair-command-arm64 }}"
+        fi
+        cibw_before_all="$cibw_before_all_ssh $cibw_before_all"
+
+        # set before-all-linux command to:
+        #   1. set up ssh-agent + deploy key for cloning private repos
+        #   2. store internal pypi credentials in .netrc
+        #   3. append `&& cibw_before_all` if it's not empty
+        if [ "${cibw_before_all}" != "" ]; then
+          echo "CIBW_BEFORE_ALL_LINUX=printf 'machine pypi.k8s.rapids.ai\n\tlogin ${{ inputs.internal-pypi-user }}\n\tpassword ${{ inputs.internal-pypi-pass }}\n' > ~/.netrc && ${cibw_before_all}" >> "${GITHUB_ENV}"
+        else
+          echo "CIBW_BEFORE_ALL_LINUX=printf 'machine pypi.k8s.rapids.ai\n\tlogin ${{ inputs.internal-pypi-user }}\n\tpassword ${{ inputs.internal-pypi-pass }}\n' > ~/.netrc" >> "${GITHUB_ENV}"
+        fi
+
+        # set repair command based on runner arch
+        echo "CIBW_REPAIR_WHEEL_COMMAND_LINUX=${auditwheel_repair_command}" >> "${GITHUB_ENV}"
+
+        # set CIBW_BUILD based on input python version e.g. 3.8 -> cp38-*
```

Comment:
Now that we're enforcing `cp38 cp39` parallelism on the level of the workflow, no need to assemble pyver this way.